include/toolchain/gcc.h

```diff
@@ -591,10 +591,18 @@ do {                                                                    \
 		1UL << (63U - __builtin_clzl(x) + 1U) : \
 		1UL << (63U - __builtin_clzl(x)))
 #else
-#define Z_POW2_CEIL(x) ((1UL << (31U - __builtin_clzl(x))) < x ?  \
-		1UL << (31U - __builtin_clzl(x) + 1U) : \
-		1UL << (31U - __builtin_clzl(x)))
+#define Z_POW2_CEIL(x) ((1UL << (31U - __builtin_clz(x))) < x ?  \
+		1UL << (31U - __builtin_clz(x) + 1U) : \
+		1UL << (31U - __builtin_clz(x)))
 #endif
 
+/**
+ * @brief Check whether or not a value is a power of 2
+ *
+ * @param x The value to check
+ * @return true if x is a power of 2, false otherwise
+ */
+#define Z_IS_POW2(x) ((x) && !((x) & ((x) - 1)))
```

Comment:
It looks like the Xtensa arch code defines its own `Z_IS_POW2` -- we should remove that.

https://github.com/zephyrproject-rtos/zephyr/blob/890f6199367953e52f2ffcf10420c98d9ce3a320/include/arch/xtensa/cache.h#L18

Also, to make it more compliant to the Zephyr/MIRSA coding guidelines:
```suggestion
#define Z_IS_POW2(x) (((x) != 0) && (((x) & ((x) - 1U)) == 0))
```

p.s. the global addition of `Z_IS_POW2` should be in a separate commit.

Reply:
Done